src/history/hs_cursor.c

```diff
@@ -62,9 +62,11 @@ __wt_hs_row_search(WT_CURSOR_BTREE *hs_cbt, WT_ITEM *srch_key, bool insert)
             hs_cursor->reset(hs_cursor);
     }
 
-    if (!leaf_found)
+    if (!leaf_found) {
+        WT_ERR(__wt_cursor_func_init(hs_cbt, true));
```

Comment:
Now that we're calling `__wt_cursor_func_init()`, I think there's no longer a need to call WT_CURSOR.reset explicitly (around like 60).